src/components/text-button.js

```diff
@@ -0,0 +1,55 @@
+AFRAME.registerComponent("text-button", {
+  schema: {
+    haptic: { type: "selector" },
+    textEl: { type: "string" },
+    textHoverColor: { type: "string" },
+    textColor: { type: "string" },
+    backgroundHoverColor: { type: "string" },
+    backgroundColor: { type: "string" }
+  },
+
+  init() {
+    this.onHover = () => {
+      this.hovering = true;
+      this.updateButtonState();
+      this.emitHapticPulse();
+    };
+    this.onHoverOut = () => {
+      this.hovering = false;
+      this.updateButtonState();
+    };
+    this.onClick = () => {
+      this.emitHapticPulse();
+    };
+    this.textEl = this.el.querySelector(this.data.textEl);
```

Comment:
Man we really should push to add a relative selector type to aframe, we end up doing this way too often.